### PR TITLE
Fix for thermostat operating mode

### PR DESCRIPTION
**Pull request by DannyDeGaspari**
Thermostat operating mode not handled correctly. The modes are : disabled, manual, automatic and sleep timer. Whenever the preset is changed (safe, night, day, comfort) the operating mode is forced to go to automatic mode.
This fix retains the operating mode when the preset is changed.

@Cereal2nd, I would also like to add following parameters to be available in home assistant:
- thermostat operating mode (disabled, manual, automatic and sleep timer)
- module program mode (none, winter, summer, holiday)

But the Climate entity does not have such properties, what would be the best way to add these parameters ? Is adding 2 extra channels to the module a good idea ? One for the operating mode and one for the program mode.

### CODE REVIEW

**Review by Cereal2nd — COMMENTED** [review]
Does every module that supports the modulestqtusmessage2 have the sleep timer value?
Same question for the tempsensorstatusmessage. Is this guaranteed to be like this for every received message?

Plus we are trying to improve the testing coverage. Could you write a couple of simple Testcases that implement the changes in this PR?

**Comment by DannyDeGaspari** [comment]
> Does every module that supports the modulestqtusmessage2 have the sleep timer value?

I'll check for all modules that are listed in this message.

> Same question for the tempsensorstatusmessage.

I did only a bug fix in here and did not add new code or functionality.

> Could you write a couple of simple Testcases that implement the changes in this PR?

I'll try, but I will have to figure out how to do this for this specific message.

**Comment by Cereal2nd** [comment]
I deed it's only the modulestatusmessage2 that needs checking

For the tests there are some good simple examples in there, the one for the modulename is a good example and is the same idea as this one.

**Comment by Cereal2nd** [comment]
thanks for adding the testcases, we are almost there.

About the question in hass, i think we should do these things with extra state attributes

File: velbusaio/messages/module_status.py [hunk 1]
```diff
@@ -141,14 +152,11 @@ def data_to_binary(self):
...
 register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBGPO")
 register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBGPOD")
+register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBGPOD-2")
 register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBELO")
 register_command(COMMAND_CODE, ModuleStatusMessage2, "VMB7IN")
-register_command(COMMAND_CODE, ModuleStatusMessage2, "VMB4DC")
-register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBDMI")
-register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBDMI-R")
-register_command(COMMAND_CODE, ModuleStatusMessage2, "VMBDME")
-register_command(COMMAND_CODE, ModuleStatusMessage2, "VMB1RYS")
```
Comment (Cereal2nd) [inline velbusaio/messages/module_status.py:150]: These modules, don't they ned this modulemessage?

**Review by Cereal2nd — CHANGES_REQUESTED** [review]

File: velbusaio/messages/module_status.py (hunk 1 above)
Comment (Cereal2nd) [reply velbusaio/messages/module_status.py:150]: do you own some of thes emodules?
i checked the docs and it seems to be correct, and i tested it VMB1RYS, but thats the only module i own from that list

Comment (DannyDeGaspari) [reply velbusaio/messages/module_status.py:150]: I only have the VMB4DC from this list. It doesn't respond to module status request.

**Review by Cereal2nd — APPROVED** [review]

**Comment by Cereal2nd** [comment]
Thanks for the help

**Comment by DannyDeGaspari** [comment]
> About the question in hass, i think we should do these things with extra state attributes

Does this require changes on the Hass side ? If yes, can you help here or at least guide me into it ?

BTW: during my check on the ModuleStatusMessage I noticed there is still another message variant needed for the VMBGP4PIR module. I own such a module and will implement support for it.

**Comment by Cereal2nd** [comment]
Yes it requires changes on the Hass side.

So what attributes exactly and how can we get these from the current channel.

If I know this it's easy.

**Comment by DannyDeGaspari** [comment]
There are 2 attributes:
- "Operating mode": stored in channel._cstatus variable and get from     
`def get_climate_mode(self) -> str:`
No set function defined yet. Should I create it ?
- "Selected program": only decoded in the message itself as a first step, further implementation needed.

**Comment by DannyDeGaspari** [comment]
@Cereal2nd, Just to let you know that I'm still continuing with further implementations of the VMBGP4PIR module, the thermostat run mode and selected program. But did not had too much spare time past week...

**Pull request merged**